embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceCacheService.kt

```diff
@@ -162,13 +162,68 @@
         return success
     }
 
-    override fun listFilenamesByPrefix(prefix: String): List<String> { // TODO: locking?
-        return storageService.listFiles { _, name ->
-            name.startsWith(EMBRACE_PREFIX + prefix)
-        }.map { file -> file.name.substring(EMBRACE_PREFIX.length) }
+    override fun normalizeCacheAndGetSessionFileIds(): List<String> {
+        val sessionFileNames = storageService
+            .listFiles { _, name -> name.startsWith(EMBRACE_PREFIX + EmbraceDeliveryCacheManager.SESSION_FILE_PREFIX) }
+            .map { file -> file.name.substring(EMBRACE_PREFIX.length) }
+
+        // Need to sweep the files directory to clear the temp files left over due to the app process crashing at an inopportune time
+        // - Files that contain the latest version of a session (i.e. suffixed of "-new") will be renamed to the proper expected name
+        // - Files that were to be deleted in anticipation of a new version but weren't will be delete now (i.e. suffix "-old")
+        // - A file named "last_session.json" used by older versions of the SDK that need to be renamed in the new format
+        //
+        // Since we are dealing with old sessions files, the current process will not modify them outside of this thread.
+        // As such, it is safe to do it on this thread.
+
+        val properSessionFileIds = mutableListOf<String>()
+        sessionFileNames.forEach { filename ->
+            if (filename.endsWith(OLD_COPY_SUFFIX)) {
+                if (!deleteFile(filename + OLD_COPY_SUFFIX)) {
```

Comment:
There's a bug here that was fixed in the subsequent PR - the delete doesn't happen because the suffix was already in the file name.  Adding a more thorough check of the underlying state of the file system in the test there.